diff/apply/apply_linux.go

```diff
@@ -50,7 +53,18 @@ func apply(ctx context.Context, mounts []mount.Mount, r io.Reader) error {
 			opts = append(opts, archive.WithParents(parents))
 		}
 		_, err = archive.Apply(ctx, path, r, opts...)
+		if err == nil && sync {
+			err = doSyncFs(path)
+		}
 		return err
+	case sync && len(mounts) == 1 && mounts[0].Type == "bind":
+		defer func() {
+			if retErr != nil {
+				return
+			}
+
+			retErr = doSyncFs(mounts[0].Source)
+		}()
```

Comment:
Just for my knowledge here, we need this logic because `umount` syscall does not flush the cache for bind mounts?

Reply:
based on test result, yes.

The alpine image has only one layer. Based on the following test, the file cache is gone after reboot. 

https://github.com/fuweid/go-dmflakey/blob/08b32bc47733b5be82ceb3e7460d250de83112f9/contrib/test/containerd/issue5854_test.go#L32